generators/server/templates/src/main/resources/config/_application.yml

```diff
@@ -186,19 +189,31 @@ security:
     oauth2:
         resource:
             filter-order: 3
-    <%_ } else if (authenticationType === 'oauth2') { _%>
+    <%_ } _%>
+    <%_ if (authenticationType === 'oauth2') {  _%>
     oauth2:
         client:
             accessTokenUri: ${oauth2.issuer}/protocol/openid-connect/token
             userAuthorizationUri: ${oauth2.issuer}/protocol/openid-connect/auth
+            <%_ if (applicationType === 'gateway') { _%>
             clientId: web_app
             clientSecret: web_app
+            tokenName: access_token
             clientAuthenticationScheme: form
             scope: openid profile email
+            <%_ } _%>
+            <%_ if (applicationType === 'microservice') { _%>
+            clientId: internal
+            clientSecret: internal
+            authenticationScheme: header
+            clientAuthenticationScheme: header
+            <%_ } _%>
         resource:
             userInfoUri: ${oauth2.issuer}/protocol/openid-connect/userinfo
             tokenInfoUri: ${oauth2.issuer}/protocol/openid-connect/token/introspectr
             preferTokenInfo: false
+            jwt:
+                keyUri: ${oauth2.issuer}
```

Comment:
These lines aren't needed for a monolith, so should only be included for microservice architecture.